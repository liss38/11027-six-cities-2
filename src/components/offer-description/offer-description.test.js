import React from 'react';
import renderer from 'react-test-renderer';
import OfferDescription from './offer-description.jsx';

it(`OfferDescription rendered correctly`, () => {
  const tree = renderer
    .create(<OfferDescription
      offer={{
        title: `Beautiful &amp; luxurious apartment at great location`,
        type: `apartment`,
        price: 50,
        rating: 4.5,
        isPremium: true,
        isFavorite: false,
        host: {
          isSuper: true,
          name: `Angelina`,
          avatarUrl: `img/avatar-angelina.jpg`,
        },
        description: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.`,
        goods: [`Heating`, `Kitchen`, `Cable TV`, `Washing machine`, `Coffee machine`, `Dishwasher`],
        maxAdults: 14,
        bedrooms: 5,
      }}
    />)
    .toJSON();

  expect(tree).toMatchSnapshot();
});
