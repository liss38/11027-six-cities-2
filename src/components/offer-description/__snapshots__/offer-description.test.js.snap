// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OfferDescription rendered correctly 1`] = `
Array [
  <div
    className="property__mark"
  >
    <span>
      Premium
    </span>
  </div>,
  <div
    className="property__name-wrapper"
  >
    <h1
      className="property__name"
    >
      Beautiful &amp; luxurious apartment at great location
    </h1>
    <button
      className="property__bookmark-button button"
      type="button"
    >
      <svg
        className="property__bookmark-icon"
        height="33"
        width="31"
      />
      <span
        className="visually-hidden"
      >
        To bookmarks
      </span>
    </button>
  </div>,
  <div
    className="property__rating rating"
  >
    <div
      className="property__stars rating__stars"
    >
      <span
        style={
          Object {
            "width": "90%",
          }
        }
      />
      <span
        className="visually-hidden"
      >
        Rating
      </span>
    </div>
    <span
      className="property__rating-value rating__value"
    >
      4.5
    </span>
  </div>,
  <ul
    className="property__features"
  >
    <li
      className="property__feature property__feature--entire"
    >
      Entire place
    </li>
    <li
      className="property__feature property__feature--bedrooms"
    >
      5
       Bedrooms
    </li>
    <li
      className="property__feature property__feature--adults"
    >
      Max 
      14
       adults
    </li>
  </ul>,
  <div
    className="property__price"
  >
    <b
      className="property__price-value"
    >
      €
      50
    </b>
    <span
      className="property__price-text"
    >
       night
    </span>
  </div>,
  <div
    className="property__inside"
  >
    <h2
      className="property__inside-title"
    >
      What's inside
    </h2>
    <ul
      className="property__inside-list"
    >
      <li
        className="property__inside-item"
      >
        Heating
      </li>
      <li
        className="property__inside-item"
      >
        Kitchen
      </li>
      <li
        className="property__inside-item"
      >
        Cable TV
      </li>
      <li
        className="property__inside-item"
      >
        Washing machine
      </li>
      <li
        className="property__inside-item"
      >
        Coffee machine
      </li>
      <li
        className="property__inside-item"
      >
        Dishwasher
      </li>
    </ul>
  </div>,
  <div
    className="property__host"
  >
    <h2
      className="property__host-title"
    >
      Meet the host
    </h2>
    <div
      className="property__host-user user"
    >
      <div
        className="property__avatar-wrapper property__avatar-wrapper--pro user__avatar-wrapper"
      >
        <img
          alt="Host avatar"
          className="property__avatar user__avatar"
          height="74"
          src="img/avatar-angelina.jpg"
          width="74"
        />
      </div>
      <span
        className="property__user-name"
      >
        Angelina
      </span>
      <span
        className="property__user-status"
      >
        Pro
      </span>
    </div>
    <div
      className="property__description"
    >
      <p
        className="property__text"
      >
        A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.
      </p>
    </div>
  </div>,
]
`;
